File: app/src/main/java/line/javafoundation/producer_consumer/Consumer.java
package line.javafoundation.producer_consumer;


public class Consumer implements Runnable {
    private Storage<Phone> storage;

    public Consumer(Storage<Phone> storage) {
        this.storage = storage;
    }

    public void run() {
        for (int i = 0; i < 20; i++) {
            storage.consume();
            try {
                Thread.sleep(1000);
//                每隔1000毫秒消费一个
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
    }
}


 

File: javalib/src/main/java/com/yin/javalib/producer_consumer/Consumer.kt
package com.yin.javalib.producer_consumer

class Consumer(private val storage: Storage<Phone>) : Runnable {
    override fun run() {
        for (i in 0..19) {
            storage.consume()
            try {
                Thread.sleep(1000)
                //                每隔1000毫秒消费一个
            } catch (e: InterruptedException) {
                e.printStackTrace()
            }
        }
    }

}